proto/tendermint/abci/types.pb.go

```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-gogo. DO NOT EDIT.
 // source: tendermint/abci/types.proto
 
-package types
+package abci
```

Comment:
One of the biggest challenges I've experienced is this. It's something small but I have no idea how to get it right with buf.

So right now the workflow for the `proto/tendermint/abci/types.proto` file is:
1. Generate all the Go stubs with a `go_package_prefix` of `github.com/tendermint/tendermint/proto`, but use `github.com/tendermint/tendermint/abci/types` as the package just for the `proto/tendermint/abci/types.proto` file
2. Move the generated `proto/tendermint/abci/types.pb.go` into the `abci` directory

I can think of ways to do this with buf as I currently understand it, but it would involve separating the `abci/types.proto` file out into its own module, reconfiguring all the other protos and buf accordingly.

Reply:
Here's the equivalent functionality using `protoc`: https://github.com/tendermint/tendermint/blob/d8b738f547f59aa991c9c0b7332d2a1fa52d0823/Makefile#L95

Reply:
This is one of those things that `protoc` allows that really shouldn't be done in our opinion - the goal would be to generate to an equivalent directory structure as your `.proto` files are laid out, across all languages (and preferably not to the same directory as your `.proto` files).

The best case here would be that you can change the generated output location globally - I'd suggest `/gen` instead of `/proto` for everything but `types.proto`, and `/` for `types.proto. What you might be doing here is that `types.proto` is your public API, and everything else is your private API - is this right? If this is the case, it actually would make logical sense for them to be two separate modules. Can discuss 1-1 if you'd like.